content/articles/android-autologout-kotlin/index.md

```diff
@@ -0,0 +1,419 @@
+        //if the user is still inactive, show the dialog
+        /*reset the number of clicks to zero to start
+        the detection  incase the dialog is dismissed*/
+        if (isActive == false) {
+            noOfClicks = 0
+            val displayDialog = AlertDialog.Builder(this)
+            displayDialog.apply {
+                setTitle("Autologout")
+                setMessage("It seems you have you have been away for a while. Would you like us to sign you out?")
+                setPositiveButton("Yes, Sign out") { _, _ ->
+
+                }
+                setNegativeButton("Keep me in") { _, _ ->
+
+                }
+            }.create().show()
+        } else {
+            /*if active, set the clicks to zero also
+            to allow the detection to start as the user clicks/taps the screen
+             */
+            noOfClicks = 0
+        }
+    }
+```
+
+If the user is still inactive, show the dialog and reset the number of clicks to 0 to start the detection. If the user and gestures dismiss, the dialog is detected. If active, set the clicks to 0 again to allow the detection to start as the user clicks/taps the screen.
+
+> View more about dialogs in [this](https://www.section.io/engineering-education/getting-started-with-dialogs-in-android-kotlin/) EngEd article.
+
+That's it in a few paragraphs. 
+
+**Note that we have to do some touches on the screen for the methods to start firing.**
+
+If you do not want to wait until someone touches the screen necessarily, you can call the `startDetection()` method in the Activity's `onCreate()` method too after instantiating the gesture detector.
+
+### Layout XML(activity_main.xml) file
+This is the XML code. It contains a TextView displaying "*Hello there. You will be logged out after 15 seconds*".
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
+        xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:tools="http://schemas.android.com/tools"
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        tools:context=".MainActivity">
+    <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="Hello there. You will be logged out after 15 seconds"
+            android:id="@+id/sampleTxt"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintLeft_toLeftOf="parent"
+            app:layout_constraintRight_toRightOf="parent"
+            app:layout_constraintTop_toTopOf="parent" android:paddingLeft="15dp" android:paddingRight="15dp"/>
+
+</androidx.constraintlayout.widget.ConstraintLayout>
+```
+
+Run the app and do some touches, scrolls, and wait 15 seconds to check the feature. Also, do the same before the seconds elapse and check. Change the numbers and check too.
```

Comment:
Hey @vinstex .
Maybe you can link the repo that will contain this project's code here and a downloadable sample APK for this project linked which you can host on Google drive.